alerts/src/commonMain/kotlin/Alerts.kt

```diff
@@ -189,13 +189,34 @@ abstract class BaseAlertBuilder {
      */
     fun addActions(actions: List<Alert.Action>) = apply { this.actions.addAll(actions) }
 
+    /**
+     * Builds an alert using DSL syntax
+     *
+     * @param initialize
+     * @return
+     */
+    fun alert(initialize: BaseAlertBuilder.() -> Unit): AlertInterface {
+        reset()
+        initialize()
+        return create()
+    }
+
     /**
      * Adds an [action] to the alert
      *
      * @param action The action object
      */
     private fun addAction(action: Alert.Action) = apply { this.actions.add(action) }
 
+    /**
+     * Reset builder into initial state
+     */
+    private fun reset() = apply {
```

Comment:
Bit unusual syntax here with `apply` and elsewhere. Why not just 

`private fun reset() { 
    title = ...
}`

Am I missing something?

Reply:
`apply` will return `this`, then we have concatenation for multiple builder function calls.

Reply:
ah makes sense